hudi-common/src/main/java/org/apache/hudi/common/util/FileIOUtils.java

```diff
@@ -91,4 +94,55 @@ public static void writeStringToFile(String str, String filePath) throws IOExcep
     out.flush();
     out.close();
   }
+
+  /**
+   * Closes a {@link Closeable}, with control over whether an {@code IOException} may be thrown.
+   * @param closeable the {@code Closeable} object to be closed, or null,
+   *      in which case this method does nothing.
+   * @param swallowIOException if true, don't propagate IO exceptions thrown by the {@code close} methods.
+   *
+   * @throws IOException if {@code swallowIOException} is false and {@code close} throws an {@code IOException}.
+   */
+  public static void close(@Nullable Closeable closeable, boolean swallowIOException)
+      throws IOException {
+    if (closeable == null) {
+      return;
+    }
+    try {
+      closeable.close();
+    } catch (IOException e) {
+      if (!swallowIOException) {
+        throw e;
+      }
+    }
+  }
+
+  /** Maximum loop count when creating temp directories. */
+  private static final int TEMP_DIR_ATTEMPTS = 10000;
+
+  /**
+   * Create a Temporary Directory.
+   * @return {@code File}
+   */
+  public static File createTempDir() {
```

Comment:
this is mostly used by tests, right?  I think we should just replace those with Junit temporary folder or use `Files.createTempDirectory()` from jdk here instead of creating it by hand.. This code and the retry loops can be simplified to 1 line, java will set up hooks to delete the directories on exit etc, if we do that..